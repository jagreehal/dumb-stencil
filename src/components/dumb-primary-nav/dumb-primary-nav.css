:host {
  display: flex;
  margin-left: auto;
}

.c-primary-nav {
  align-items: center;
  display: inherit;
}

.c-primary-nav__list {
  display: flex;
  list-style: none;
  margin: 0;
  padding: 0;
}

.c-primary-nav__item {
  margin-right: 2rem;
}
