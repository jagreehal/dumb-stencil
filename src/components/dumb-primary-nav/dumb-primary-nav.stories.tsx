import '@stencil/core';

import { storiesOf } from '@storybook/html';

const stories = storiesOf('Navigation/PrimaryNav', module);

stories.add('Default', () => {
  const listItems = [
    {
      href: '#',
      text: 'Nav Item 1'
    },
    {
      href: '#',
      text: 'Nav Item 2'
    },
    {
      href: '#',
      text: 'Nav Item 3'
    },
    {
      href: '#',
      text: 'Nav Item 4'
    }
  ];

  return `<dumb-primary-nav list-items='${JSON.stringify(listItems)}'>
    </dumb-primary-nav>`;
});
