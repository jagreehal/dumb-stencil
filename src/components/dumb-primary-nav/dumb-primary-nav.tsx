import { Component, Prop } from '@stencil/core';

export interface PrimaryNavItem {
  href: string;
  text: string;
}

@Component({
  tag: 'dumb-primary-nav',
  styleUrl: 'dumb-primary-nav.css',
  shadow: true
})
export class DumbPrimaryNav {
  @Prop() listItems: string | PrimaryNavItem[];
  @Prop() styleModifier: string;

  render() {
    const _listItems: PrimaryNavItem[] = Array.isArray(this.listItems)
      ? this.listItems
      : JSON.parse(this.listItems);
    return (
      <nav class={`c-primary-nav ${this.styleModifier || ''}`}>
        <ul class="c-primary-nav__list">
          {_listItems.map(listItem => {
            return (
              <li class="c-primary-nav__item">
                <a href={listItem.href}>{listItem.text}</a>
              </li>
            );
          })}
        </ul>
      </nav>
    );
  }
}
