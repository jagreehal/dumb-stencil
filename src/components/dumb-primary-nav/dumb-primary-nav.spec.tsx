import { TestWindow } from '@stencil/core/dist/testing';
import { DumbPrimaryNav } from './dumb-primary-nav';

describe('dumb-primary-nav', () => {
  let element: HTMLElement;
  let window;
  const listItems = [
    {
      "href": "#",
      "text": "Nav Item 1"
    },
    {
      "href": "#",
      "text": "Nav Item 2"
    },
    {
      "href": "#",
      "text": "Nav Item 3"
    },
    {
      "href": "#",
      "text": "Nav Item 4"
    }
  ];

  beforeEach(async () => {
    window = new TestWindow();
    element = await window.load({
      components: [DumbPrimaryNav],
      html: `<dumb-primary-nav list-items='${JSON.stringify(listItems)}'></dumb-primary-nav>`
    });
  });

  it('should render four list elements', async () => {
    expect(element.querySelectorAll('li').length).toBe(4);
  });

  it('should match snapshot', async () => {
    expect(element).toMatchSnapshot();
  });
});
