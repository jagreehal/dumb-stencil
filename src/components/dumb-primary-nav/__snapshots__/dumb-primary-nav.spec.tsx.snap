// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dumb-primary-nav should match snapshot 1`] = `
<dumb-primary-nav
  class="hydrated"
  list-items="[{\\"href\\":\\"#\\",\\"text\\":\\"Nav Item 1\\"},{\\"href\\":\\"#\\",\\"text\\":\\"Nav Item 2\\"},{\\"href\\":\\"#\\",\\"text\\":\\"Nav Item 3\\"},{\\"href\\":\\"#\\",\\"text\\":\\"Nav Item 4\\"}]"
>
  
  <nav
    class="c-primary-nav"
  >
    <ul
      class="c-primary-nav__list"
    >
      <li
        class="c-primary-nav__item"
      >
        <a
          href="#"
        >
          Nav Item 1
        </a>
      </li>
      <li
        class="c-primary-nav__item"
      >
        <a
          href="#"
        >
          Nav Item 2
        </a>
      </li>
      <li
        class="c-primary-nav__item"
      >
        <a
          href="#"
        >
          Nav Item 3
        </a>
      </li>
      <li
        class="c-primary-nav__item"
      >
        <a
          href="#"
        >
          Nav Item 4
        </a>
      </li>
    </ul>
  </nav>
</dumb-primary-nav>
`;
